### PR TITLE
Comment Moderation: when Next is tapped on the snackbar, show the next comment.

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #17290 

When `Next` is tapped on the comment moderation confirmation snackbar, the comment details view is updated with the next comment in the current filter. If the currently displayed comment is the last one, `Next` is not displayed on the snackbar.

To test:
- Enable the `newCommentDetail` feature.
- Go to My Site > Comments.
- Select a comment and moderate it. Stay on the details view.
  - Tap `Next` on the snackbar.
  - Verify the comment details view is updated with the next comment in the list.
- Select the last comment in a list and moderate it. Stay on the details view.
  - Verify the confirmation snackbar does not have a `Next` button.
 
NOTE: The comment content field height is not updated correctly when the next comment has less content than the prior one. The same problem occurs when editing a comment to have less content. As that is not related to this change, I'll leave that to be addressed later.

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Comments/CommentsViewController.m [hunk 1]
```diff
@@ -378,6 +403,21 @@ - (void)deleteComment:(Comment *)comment
...
+// When `Next` is tapped on the comment moderation confirmation snackbar,
+// find the next comment in the list and update comment details with it.
+- (void)showNextComment
+{
+    NSIndexPath *nextIndexPath = [NSIndexPath indexPathForRow:self.displayedCommentIndexPath.row + 1
+                                                    inSection:self.displayedCommentIndexPath.section];
+    
+    if (![self indexPathIsValid:nextIndexPath] || !self.commentDetailViewController) {
+        return;
+    }
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Comments/CommentsViewController.m]: I've tested this, and it seems that there's a little bug here; the Next button does nothing when the comment is on the last row of a section. For example, tapping Next after moderating the last comment on the "Older than a week" section should load the first comment in the "Older than a month" section.

**Review by dvdchr — CHANGES_REQUESTED** [review]
Overall, the code looks great 👍 . Also, I've verified that almost everything works well, except for one little bug 🐛 .

File: WordPress/Classes/ViewRelated/Comments/CommentsViewController.m (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Comments/CommentsViewController.m]: Oh that's funny. I noticed `Next` didn't show at all on the last row of section, so I fixed that. Clearly I didn't check to make sure it actually worked though. :facepalm:  Thank you I'll take a look!

**Comment by ScoutHarris** [comment]
Hey @dvdchr . `Next` on the last row of a section should work now. Ready for another look please!

**Review by dvdchr — APPROVED** [review]

**Comment by dvdchr** [comment]
LGTM! :shipit: 

> NOTE: The comment content field height is not updated correctly when the next comment has less content than the prior one. The same problem occurs when editing a comment to have less content. As that is not related to this change, I'll leave that to be addressed later.

Thanks! I'll take a look at this tomorrow.

**Comment by ScoutHarris** [comment]
Thanks @dvdchr !

**Pull request merged**